src/app/shared/planet-language.component.ts

```diff
@@ -21,4 +22,12 @@ export class PlanetLanguageComponent implements OnInit {
       language => language.shortCode !== this.currentLanguage.shortCode);
   }
 
+  getRouterUrl(language) {
+    const newRoute = '/' + language.shortCode + this.router.url;
+    if (this.router.url.indexOf('/' + this.currentLanguage.shortCode + '/') === 0) {
```

Comment:
language code is not part of router.url as we are changing basehref. So removing this condition

Reply:
perfect. I wasn't aware about the basehref changes